rfcs/bring-ts.md

```diff
@@ -0,0 +1,248 @@
+ha?
+
+### Consuming the library
+
+OK, let's use this library in our Wing code.
+
+```js
+bring colors;
+
+print("${colors.Global.green("hello")}, ${colors.Colors.blue("world")}!")
+```
+
+As you can see, since Wing doesn't support global functions, the `Global` wrapper class was
+generated and includes a static method for each public API exported from `colors`.
+
+Similarly to turning global functions to static methods, `ts2wing` applies other heuristics to
+create an [adaptation layer](#type-system-adapters) between the type systems.
+
+### Escape hatch
+
+As you might have expected, there could be cases where `ts2wing` won't be able to create an adapter
+for a certain TypeScript library. It will take us time to establish all the patterns and in parallel
+evolve the Wing type system.
+
+If you run `ts2wing` with the switch `--unpacked` like so:
+
+```sh
+$ ts2wing edit colors
+OK, "colors" is ready to be edited under "vendor/colors.wlib.src".
+After you are done, make sure to run: ts2wing colors
+```
+
+Now, go ahead and make any modifications you like to `vendor/colors.wlib.src` and after
+you are finished, run:
+
+```sh
+$ ts2wing colors
+Found vendor/colors.wlib.src
+Processing type definitions (index.d.ts)...
+Generating Wing wrapper classes...
+Compiling with JSII...
+Generating API.md...
+Packaging Wing Library to vendor/colors.wlib
+Cleaning vendor/colors.wlib.src
+```
+
+This is it, now the updated `colors.wlib` will include your changes.
+
+### Type system adapters
+
+It is not possible to express every type in TypeScript within the Wing type system. This is because:
+
+1. Wing is young and it will take the type system time to evolve; 
+2. The design goals of Wing and TypeScript are different. TypeScript is designed to be able to
+   represent every possible type that can be expressed in JavaScript, while Wing is designed to be a
+   great language for cloud application programming.
+
+#### Union Types
+
+A good example is **type unions**. Since in JavaScript is dynamic, it is possible for a value to
+have multiple possible types. In TypeScript this is expressed through unions.
+
+Let's take an example:
+
+```ts
+declare class Foo {
+  constructor(value: string | number);
+  bar():  string[] | string | false;
+}
+```
+
+The constructor of `Foo` accepts a single argument that can be either a `string` or a `number`.
+
+This is all nice, but at the moment we believe that supporting type unions is not really required
+for building great cloud applications. In fact, we think that type unions are a potential source of
+bugs and complexity, and we want Wing to be easy to learn and use and scale across large teams in
+big companies. To that end, we don't support type unions (and we reserve the right to change our
+minds).
+
+To bridge this gap, the `ts2wing` tool will generate a "union-like class" in for each type union in
+the imported library. The class will have a set of static factory methods, one for each type in the
+union:
```

Comment:
As a TS fan, this reads as "We think type unions are bad so instead of having them, we're doing something worse". 
It's fair to say we aren't convinced they're important to have in Wing and that it maybe is too difficult to implement right now so we're doing something easier.

Basically, the following specific part doesn't seem helpful, because it's strange to criticize the existence of something while attempting to mimic it:
> In fact, we think that type unions are a potential source of bugs and complexity, and we want Wing to be easy to learn and use and scale across large teams in big companies.